### PR TITLE
docs: Update main crate docstring

### PR DESCRIPTION
**Pull request by max-sixty**
I think `prql-compiler-macros` is now published. And I refined the style a bit.

Does the `compile-files` example need a macro within it? It complicates the example slightly — we're saying "if you want to compile files, then use `prql-compiler`, rather than `prql-compiler-macros` but write your own macro"??

### CODE REVIEW

File: prql-compiler/src/lib.rs [hunk 1]
```diff
@@ -49,23 +47,21 @@
 //!     # }
 //!     ```
 //!
-//! - ... compile PRQL queries to SQL at build time.
+//! - Compile PRQL queries to SQL at build time.
 //!
-//!     Use `prql-compiler-macros` crate (unreleased), which can be used like
-//!     this:
+//!     Use the `prql-compiler-macros` crate; for example:
```
Comment (aljazerzen) [inline prql-compiler/src/lib.rs]:
```suggestion
//!     For inline strings, use the `prql-compiler-macros` crate; for example:
```

File: prql-compiler/src/lib.rs [hunk 2]
```diff
@@ -49,23 +47,21 @@
...
 //!     ```ignore
 //!     let sql: &str = prql_to_sql!("from albums | select {title, artist_id}");
 //!     ```
 //!
-//! - ... compile .prql files to .sql files at build time.
+//!     Alternatively, to compile `.prql` files to `.sql` files at build time,
+//!     call `prql-compiler` from `build.rs`. See [this example
+//!     project](https://github.com/PRQL/prql/tree/main/prql-compiler/examples/compile-files).
```
Comment (aljazerzen) [inline prql-compiler/src/lib.rs]:
```suggestion
//!     For compiling whole files (`.prql` to `.sql`), call `prql-compiler`
//!     from `build.rs`.
//!     See [this example project](https://github.com/PRQL/prql/tree/main/prql-compiler/examples/compile-files).
```

**Review by aljazerzen — APPROVED** [review]
Nice.

File: prql-compiler/src/lib.rs (hunk 2 above)
Comment (max-sixty) [reply prql-compiler/src/lib.rs]: Good suggestions!

**Pull request merged**